### PR TITLE
React 17.0.1

### PR DESCRIPTION
**Pull request by qmonmert**
React 17.0.1 🚀

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by DanielFran** [comment]
@qmonmert Better to add it now on in the package.json, so dependabot can trigger new updates?

**Comment by qmonmert** [comment]
@DanielFran you're right, done

**Review by DanielFran — APPROVED** [review]

**Comment by DanielFran** [comment]
@mshima it seems missed by something for react to use package.json?

File: generators/client/templates/react/package.json [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {
   "devDependencies": {
+    "react": "17.0.1",
+    "react-dom": "17.0.1",
```
Comment (mshima) [inline generators/client/templates/react/package.json:4]:
```suggestion
  "dependencies": {
    "react": "17.0.1",
    "react-dom": "17.0.1",
  },
  "devDependencies": {
```

**Comment by qmonmert** [comment]
thanks @mshima it works

cc @DanielFran

**Pull request merged**